### ISSUE TITLE
Masking behavior is not working with crf

### ISSUE THREAD

**Issue by r-wheeler** [tensorflow/addons#1443]
**System information**
- debian
- 2.1, pip
- TensorFlow-Addons version and how it was installed (source or binary): 0.6.0
- Python version: 3.6.8
- Is GPU used? (yes/no): no 

**Describe the bug**

crf is not correctly masking the predicted decode_tags with respect to sequence_lengths as compared to the previous implementation in tensorflow contrib. 

**Code to reproduce the issue**

## In tensorflow 1.15 using contrib ##
```
import tensorflow as tf 
import numpy as np

tf.__version__

np.random.seed(1)

inputs = tf.constant(np.random.normal(1,size=[2, 10, 5]), dtype=tf.float32)
transition_params = tf.constant(np.ones([5, 5], dtype=np.float32))
sequence_lengths = tf.constant(np.array([5,2], dtype=np.int32))

out = tf.contrib.crf.crf_decode(inputs, transition_params, sequence_lengths)

with tf.Session() as sess:
    sess.run(tf.initialize_all_variables())
    print(sess.run(out))

Out[1]:(array([[0, 1, 0, 4, 1, 0, 0, 0, 0, 0],
       [0, 2, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=int32), array([15.558804 ,  4.2312727], dtype=float32))
```
with the sequence_lengths set to [5,2] respectively, all values beyond these lengths are set to 0  

## In tensorflow 2.1.0 using tensorflow_addons ##

```
import tensorflow as tf 
import tensorflow_addons as tfa
import numpy as np 

np.random.seed(1)
inputs = tf.constant(np.random.normal(1,size=[2, 10, 5]), dtype=tf.float32)
transition_params = tf.constant(np.ones([5, 5], dtype=np.float32))
sequence_lengths = tf.constant(np.array([5,2], dtype=np.int32))

out = tfa.text.crf_decode(inputs, transition_params, sequence_lengths)
print(out)
Out[1]: (<tf.Tensor: id=6982, shape=(2, 10), dtype=int32, numpy=
 array([[0, 1, 0, 4, 1, 4, 4, 4, 4, 4],
        [0, 2, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=int32)>,
 <tf.Tensor: id=6984, shape=(2,), dtype=float32, numpy=array([15.558804 ,  4.2312727], dtype=float32)>
``` 

the first row of the output tensor is not masked after a sequence_length of 5. This is reproducible across various inputs and sequence lengths. 

**Other info / logs**

Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by tc-wolf** [comment]
I think that the difference is that in 1.15, [dynamic_rnn](https://github.com/tensorflow/tensorflow/blob/5d80e1e8e6ee999be7db39461e0e79c90403a2e4/tensorflow/contrib/crf/python/ops/crf.py#L208) was used in the `_multi_seq_fn`.  This zeros its outputs past sequence length by default [according to docs](https://github.com/tensorflow/tensorflow/blob/5d80e1e8e6ee999be7db39461e0e79c90403a2e4/tensorflow/python/ops/rnn.py#L581-L584).

To do this for the TFA version, the [RNN layer](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/crf.py#L448-L450) has to be created with `zero_output_for_mask = True` (by default this is False).

I tested this by monkeypatching the method:
![image](https://user-images.githubusercontent.com/50339167/77699354-47d72300-6f80-11ea-9c44-67410e6335d2.png)

**Comment by gabrieldemarmiesse** [comment]
Maybe @squadrick can weight in on that?

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
